### PR TITLE
Create a data model

### PR DESCRIPTION
**Pull request by Lembani**
### This PR contains the `Creation of a Data Model` for the blog app.

> A blog that shows a list of posts and empowers readers to interact with them by adding comments and liking posts.

> Highlights: 

- Created a new app.
- The project has Postgres database set up.
- Built the project schema:
    - Created and run the necessary migration files.
    - Table and column names match the ERD diagram.
        - note: photo for users table should be a link to an image
    - Foreign keys are included.
    - All columns that are foreign keys have a corresponding index.

### CODE REVIEW

File: db/migrate/20220812142716_create_posts.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class CreatePosts < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.string :title
+      t.text :text
+      t.references :user, null: false, foreign_key: true
```
Comment (rloterh) [inline db/migrate/20220812142716_create_posts.rb]: - [x] As per the given ERD diagram _user_ on the _post_ model is referenced as an _author_, similar to:
```ruby
t.references :author, foreign_key: { to_table: 'users' }
```

File: db/migrate/20220812142820_create_comments.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class CreateComments < ActiveRecord::Migration[7.0]
+  def change
+    create_table :comments do |t|
+      t.text :text
+      t.references :post, null: false, foreign_key: true
+      t.references :user, null: false, foreign_key: true
```
Comment (rloterh) [inline db/migrate/20220812142820_create_comments.rb]: - [x] As per the given ERD diagram _user_ on the _comment_ model is referenced as an _author_, similar to:
```ruby
t.references :author, foreign_key: { to_table: 'users' }
```

File: db/migrate/20220812142903_create_likes.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreateLikes < ActiveRecord::Migration[7.0]
+  def change
+    create_table :likes do |t|
+      t.references :user, null: false, foreign_key: true
```
Comment (rloterh) [inline db/migrate/20220812142903_create_likes.rb]: - [x] As per the given ERD diagram _user_ on the _like_ model is referenced as an _author_, similar to:
```ruby
t.references :author, foreign_key: { to_table: 'users' }
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Lembani, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Entities properly created


## Required Changes ♻️
_Kindly check the comments under the review._

<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/migrate/20220812142716_create_posts.rb (hunk 1 above)
Comment (Lembani) [reply db/migrate/20220812142716_create_posts.rb]: Hi @rloterh, I have updated according to the ERD as you suggested. Thanks for the feedback.

File: db/migrate/20220812142820_create_comments.rb (hunk 2 above)
Comment (Lembani) [reply db/migrate/20220812142820_create_comments.rb]: Hi @rloterh, I have updated according to the ERD as you suggested. Thanks for the feedback.

File: db/migrate/20220812142903_create_likes.rb (hunk 3 above)
Comment (Lembani) [reply db/migrate/20220812142903_create_likes.rb]: Hi @rloterh, I have updated according to the ERD as you suggested. Thanks for the feedback.

**Review by alexander16108 — APPROVED** [review]
Hi @Lembani , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lembani** [comment]
Hi @alexander16108, thanks for approving my PR. 🙏🏾

**Pull request merged**